File: Customer_Support/scripts/seed_data.py
import os
import sys
import asyncio
import random
import uuid
from datetime import datetime, timedelta
from faker import Faker
from motor.motor_asyncio import AsyncIOMotorClient
from dotenv import load_dotenv

import certifi

# Ensure we can import app modules if needed
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

load_dotenv()

MONGODB_URI = os.getenv("MONGODB_URI", "mongodb://localhost:27017")
MONGODB_DATABASE = os.getenv("MONGODB_DATABASE", "continuum_db")

fake = Faker("en_IN")

NUM_CUSTOMERS = 1000
NUM_TRANSACTIONS = 25000

LANGUAGES = ["english", "hindi", "marathi", "tamil", "telugu", "kannada", "bengali"]
MERCHANTS = ["Amazon", "Flipkart", "Zomato", "Swiggy", "Uber", "Ola", "Reliance Fresh", "DMart", "Netflix", "Spotify", "IRCTC", "MakeMyTrip"]

async def seed_data():
    print(f"Connecting to MongoDB at {MONGODB_URI}...")
    client = AsyncIOMotorClient(
        MONGODB_URI,
        tls=True,
        tlsCAFile=certifi.where(),
        tlsAllowInvalidCertificates=True
    )
    db = client[MONGODB_DATABASE]
    
    print("Clearing existing customers and transactions...")
    await db.customers.delete_many({})
    await db.transactions.delete_many({})
    
    print(f"Generating {NUM_CUSTOMERS} customers...")
    customers = []
    customer_ids = []
    for _ in range(NUM_CUSTOMERS):
        c_id = f"CUST_{uuid.uuid4().hex[:8].upper()}"
        customer_ids.append(c_id)
        
        phone = f"+91 {random.randint(6000000000, 9999999999)}"
        
        customer = {
            "_id": c_id,
            "id": c_id,
            "name": fake.name(),
            "phone_number": phone,
            "language_preference": random.choice(LANGUAGES),
            "created_at": datetime.utcnow()
        }
        customers.append(customer)
        
    await db.customers.insert_many(customers)
    print("Customers inserted.")
    
    print(f"Generating {NUM_TRANSACTIONS} transactions...")
    transactions = []
    for i in range(NUM_TRANSACTIONS):
        c_id = random.choice(customer_ids)
        t_id = f"TXN_{uuid.uuid4().hex[:12].upper()}"
        
        days_ago = random.randint(0, 365)
        t_time = datetime.utcnow() - timedelta(days=days_ago, minutes=random.randint(0, 1440))
        
        tx = {
            "_id": t_id,
            "id": t_id,
            "customer_id": c_id,
            "account_id": f"ACC_{c_id[-8:]}",
            "amount": round(random.uniform(50.0, 15000.0), 2),
            "merchant": random.choice(MERCHANTS),
            "timestamp": t_time,
            "status": "COMPLETED"
        }
        transactions.append(tx)
        
        if len(transactions) == 5000:
            await db.transactions.insert_many(transactions)
            transactions = []
            print(f"Inserted {i+1}/{NUM_TRANSACTIONS} transactions...")
            
    if transactions:
        await db.transactions.insert_many(transactions)
        
    print("All transactions inserted successfully.")
    print("Seeding complete.")

if __name__ == "__main__":
    asyncio.run(seed_data())

File: Customer_Support/scripts/seed_data.js
const { MongoClient } = require('mongodb');
const { faker } = require('@faker-js/faker');
const { v4: uuidv4 } = require('uuid');
require('dotenv').config({ path: '../.env' });

const MONGODB_URI = process.env.MONGODB_URI || "mongodb://localhost:27017";
const MONGODB_DATABASE = process.env.MONGODB_DATABASE || "continuum_db";

const NUM_CUSTOMERS = 1000;
const NUM_TRANSACTIONS = 25000;

const LANGUAGES = ["english", "hindi", "marathi", "tamil", "telugu", "kannada", "bengali"];
const MERCHANTS = ["Amazon", "Flipkart", "Zomato", "Swiggy", "Uber", "Ola", "Reliance Fresh", "DMart", "Netflix", "Spotify", "IRCTC", "MakeMyTrip"];

function randomInt(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

async function seedData() {
    console.log(`Connecting to MongoDB at ${MONGODB_URI}...`);
    const client = new MongoClient(MONGODB_URI);

    try {
        await client.connect();
        const db = client.db(MONGODB_DATABASE);
        
        console.log("Clearing existing customers and transactions...");
        await db.collection('customers').deleteMany({});
        await db.collection('transactions').deleteMany({});
        
        console.log(`Generating ${NUM_CUSTOMERS} customers...`);
        const customers = [];
        const customerIds = [];
        
        for (let i = 0; i < NUM_CUSTOMERS; i++) {
            const cId = `CUST_${uuidv4().replace(/-/g, '').substring(0, 8).toUpperCase()}`;
            customerIds.push(cId);
            
            const phone = `+91 ${randomInt(6000000000, 9999999999)}`;
            
            customers.push({
                _id: cId,
                id: cId,
                name: faker.person.fullName(),
                phone_number: phone,
                language_preference: LANGUAGES[randomInt(0, LANGUAGES.length - 1)],
                created_at: new Date()
            });
        }
        
        await db.collection('customers').insertMany(customers);
        console.log("Customers inserted.");
        
        console.log(`Generating ${NUM_TRANSACTIONS} transactions...`);
        let transactions = [];
        let totalInserted = 0;
        
        for (let i = 0; i < NUM_TRANSACTIONS; i++) {
            const cId = customerIds[randomInt(0, customerIds.length - 1)];
            const tId = `TXN_${uuidv4().replace(/-/g, '').substring(0, 12).toUpperCase()}`;
            
            const daysAgo = randomInt(0, 365);
            const tTime = new Date();
            tTime.setDate(tTime.getDate() - daysAgo);
            tTime.setMinutes(tTime.getMinutes() - randomInt(0, 1440));
            
            const amount = parseFloat((Math.random() * (15000.0 - 50.0) + 50.0).toFixed(2));
            
            transactions.push({
                _id: tId,
                id: tId,
                customer_id: cId,
                account_id: `ACC_${cId.substring(cId.length - 8)}`,
                amount: amount,
                merchant: MERCHANTS[randomInt(0, MERCHANTS.length - 1)],
                timestamp: tTime,
                status: "COMPLETED"
            });
            
            if (transactions.length === 5000) {
                await db.collection('transactions').insertMany(transactions);
                totalInserted += transactions.length;
                transactions = [];
                console.log(`Inserted ${totalInserted}/${NUM_TRANSACTIONS} transactions...`);
            }
        }
        
        if (transactions.length > 0) {
            await db.collection('transactions').insertMany(transactions);
            totalInserted += transactions.length;
            console.log(`Inserted ${totalInserted}/${NUM_TRANSACTIONS} transactions...`);
        }
        
        console.log("All transactions inserted successfully.");
        console.log("Seeding complete.");
        
    } catch (e) {
        console.error(e);
    } finally {
        await client.close();
    }
}

seedData();
